In [177]:
import pandas as pd

df = pd.read_csv('employee.csv', encoding='latin-1')

In [111]:
import numpy as np
# Calcule le nombre de valeurs manquante selon chaque feature
def compute_missing_values(df):
    list_missing_val = {}
    col = df.columns
    for i, k in enumerate(col):
        list_missing_val[k] = df[k].isna().sum()
    return list_missing_val

In [98]:
df.shape
list = compute_missing_values(df)
print("missing values")
for i in list:
    print(f"{i}-->{list[i]:d}")

missing values
NAME-->6
DEPARTMENT_NAME-->6
TITLE-->6
REGULAR-->644
RETRO-->22150
OTHER-->8423
OVERTIME-->15706
INJURED-->21096
DETAIL-->20493
QUINN_EDUCATION_INCENTIVE-->21166
TOTAL_GROSS-->6
POSTAL-->6


In [162]:
# compte le nombre de tuples vides (toutes les features à nan)
df[df.isna().all(axis=1)].shape[0]
# suppression des tuples vides
df = df.dropna(how='all')

In [163]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 22546 entries, 0 to 22545
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   NAME                       22546 non-null  object
 1   DEPARTMENT_NAME            22546 non-null  object
 2   TITLE                      22546 non-null  object
 3   REGULAR                    21908 non-null  object
 4   RETRO                      402 non-null    object
 5   OTHER                      14129 non-null  object
 6   OVERTIME                   6846 non-null   object
 7   INJURED                    1456 non-null   object
 8   DETAIL                     2059 non-null   object
 9   QUINN_EDUCATION_INCENTIVE  1386 non-null   object
 10  TOTAL_GROSS                22546 non-null  object
 11  POSTAL                     22546 non-null  object
dtypes: object(12)
memory usage: 2.2+ MB


In [178]:
col = ['REGULAR', 'RETRO', 'OTHER', 'OVERTIME', 'INJURED', 'DETAIL', 
       'QUINN_EDUCATION_INCENTIVE', 'TOTAL_GROSS', 'POSTAL']

df[col] = df[col].apply(pd.to_numeric, errors='coerce')

In [179]:
df

NAME           DEPARTMENT_NAME  \
0            Beckers,Richard  Boston Police Department   
1      McGowan,Jacqueline M.  Boston Police Department   
2             Harris,Shawn N  Boston Police Department   
3          Washington,Walter  Boston Police Department   
4          Mosley Jr.,Curtis  Boston Police Department   
...                      ...                       ...   
22547                    NaN                       NaN   
22548                    NaN                       NaN   
22549                    NaN                       NaN   
22550                    NaN                       NaN   
22551                    NaN                       NaN   

                                TITLE  REGULAR  RETRO  OTHER  OVERTIME  \
0                      Police Officer      NaN    NaN    NaN       NaN   
1                      Police Officer      NaN    NaN    NaN       NaN   
2      Police Offc Comm Serv Offc 3$8      NaN    NaN    NaN       NaN   
3                      Police Officer      NaN    NaN    NaN       NaN   
4      Police Offc Comm Serv Offc 3$8      NaN    NaN    NaN       NaN   
...                               ...      ...    ...    ...       ...   
22547                             NaN      NaN    NaN    NaN       NaN   
22548                             NaN      NaN    NaN    NaN       NaN   
22549                             NaN      NaN    NaN    NaN       NaN   
22550                             NaN      NaN    NaN    NaN       NaN   
22551                             NaN      NaN    NaN    NaN       NaN   

       INJURED  DETAIL  QUINN_EDUCATION_INCENTIVE  TOTAL_GROSS  POSTAL  
0          NaN     NaN                        NaN          NaN  2119.0  
1          NaN     NaN                        NaN          NaN  2129.0  
2          NaN     NaN                        NaN          NaN  2130.0  
3          NaN     NaN                        NaN          NaN  2368.0  
4          NaN     NaN                        NaN          NaN  2301.0  
...        ...     ...                        ...          ...     ...  
22547      NaN     NaN                        NaN          NaN     NaN  
22548      NaN     NaN                        NaN          NaN     NaN  
22549      NaN     NaN                        NaN          NaN     NaN  
22550      NaN     NaN                        NaN          NaN     NaN  
22551      NaN     NaN                        NaN          NaN     NaN  

[22552 rows x 12 columns]

In [159]:
for colonne in colonnes_a_convertir:
    print(f"Valeurs uniques dans {colonne} avant la conversion :")
    print(df[colonne].unique())


NAME  DEPARTMENT_NAME  TITLE  REGULAR  RETRO  OTHER  OVERTIME  INJURED  \
0       NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
1       NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
2       NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
3       NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
4       NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
...     ...              ...    ...      ...    ...    ...       ...      ...   
22547   NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
22548   NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
22549   NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
22550   NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   
22551   NaN              NaN    NaN      NaN    NaN    NaN       NaN      NaN   

       DETAIL  QUINN_EDUCATION_INCENTIVE  TOTAL_GROSS  POSTAL  
0         NaN                        NaN          NaN  2119.0  
1         NaN                        NaN          NaN  2129.0  
2         NaN                        NaN          NaN  2130.0  
3         NaN                        NaN          NaN  2368.0  
4         NaN                        NaN          NaN  2301.0  
...       ...                        ...          ...     ...  
22547     NaN                        NaN          NaN     NaN  
22548     NaN                        NaN          NaN     NaN  
22549     NaN                        NaN          NaN     NaN  
22550     NaN                        NaN          NaN     NaN  
22551     NaN                        NaN          NaN     NaN  

[22552 rows x 12 columns]

# Rapport

## Q2:
- La commande count() nous permet de connaître le nombre de tuple pour chaque feature. Nous avons 22546 enregistrements et 12
  colonnes (features).
- La fonction codée plus haut nous permet de connaître le nombre de valeurs manquantes. En voici la sortie:
  missing values
NAME-->6
DEPARTMENT_NAME-->6
TITLE-->6
REGULAR-->644
RETRO-->22150
OTHER-->8423
OVERTIME-->15706
INJURED-->21096
DETAIL-->20493
QUINN_EDUCATION_INCENTIVE-->21166
TOTAL_GROSS-->6
- Oui il y 6 tuples dont toutes les colonnes sont vides.


## Q3POSTAL-->6





